### PR TITLE
Windows: Re-create custom NAT networks after restart if missing from HNS

### LINKED ISSUE #43408: Windows containers does not restart after Windows Server 2019 restart.
Windows Server 2019 does not start containers automatically on reboot, even when all docker-compose are labelled as restart: always.

**Steps to reproduce the issue:**
1. Open PowerShell with admin privileges.
2. Run any docker-compose file: `docker-compose up`
```
version: "3"

services:

  postgres_db:
    image: stellirin/postgres-windows:12-1809
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
    ports:
      - "5432:5432"
```
3. Execute Restart-Computer.
When the machine starts up, execute docker ps in PowerShell with admin privileges. 

**Describe the results you received:**
docker ps shows no containers running.
Running `docker start $(docker ps -a -q -f status=exited)` returns:
```
Error response from daemon: network 325cbdb7ce3b29e0c744d76783d94bc7637293b6efce7d912669581d5af5e4eb not found
Error response from daemon: network f555844ad59f8e1e9d95a01a8fce580ac1f2a0582acb2475c218fb8aab5bd1fb not found
Error: failed to start containers: 3cf6c5c3207a, 1df07dd99139
```

**Describe the results you expected:**
docker ps should should show containers running.

**Additional information you deem important (e.g. issue happens only occasionally):**
Happens every time on Windows Server 2019.

**Output of `docker version`:**

```
Client: Mirantis Container Runtime
 Version:           20.10.9
 API version:       1.41
 Go version:        go1.16.12m2
 Git commit:        591094d
 Built:             12/21/2021 21:34:30
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Mirantis Container Runtime
 Engine:
  Version:          20.10.9
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.16.12m2
  Git commit:       9b96ce992b
  Built:            12/21/2021 21:33:06
  OS/Arch:          windows/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  cluster: Manage Mirantis Container Cloud clusters (Mirantis Inc., v1.9.0)
  registry: Manage Docker registries (Docker Inc., 0.1.0)

Server:
 Containers: 2
  Running: 0
  Paused: 0
  Stopped: 2
 Images: 2
 Server Version: 20.10.9
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: process
 Kernel Version: 10.0 17763 (17763.1.amd64fre.rs5_release.180914-1434)
 Operating System: Windows Server 2019 Standard Version 1809 (OS Build 17763.1817)
 OSType: windows
 Architecture: x86_64
 CPUs: 8
 Total Memory: 16GiB
 Name: WIN-T721FCB2TP1
 ID: NDM2:5C34:GAV5:QDFF:TF52:327Y:HQUW:EZU5:ROH3:4T5B:QPML:KZ7H
 Docker Root Dir: C:\ProgramData\docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows Server 2019 (Version 1809)

Server on is a VM on a physical machine located inside my office building.  

Very very similar issue to https://github.com/moby/moby/issues/39623.  However, it seems like my network is disappearing on restart.

### PR DESCRIPTION
**Pull request by olljanat**
**- What I did**
Revisited on logic added by #20478 to correctly handle non-default NAT networks on Windows after server restart.
Closes #43408

**- How I did it**
Added logic which will re-create those networks with same settings after deleting orphan.

**- How to verify it**
1. Create custom network and start container to it:
```powershell
docker network create --driver nat test
docker run --net test --restart always --name test -d mcr.microsoft.com/windows/servercore:ltsc2022 ping -t 127.0.0.1
```
2. Restart server and check that test container is running.

Test binary can be found from https://filebin.net/t5nyuqjjx7un8lpd/dockerd.exe
However note that only networks which are not already missing/broken before switching to new binary version will works correctly because current behavior is that those networks get totally removed from Docker.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]

File: daemon/daemon_windows.go [hunk 1]
```diff
@@ -264,6 +264,35 @@ func (daemon *Daemon) initNetworkController(activeSandboxes map[string]interface
...
+				netOption := map[string]string{}
+				for k, v := range v.Info().DriverOptions() {
+					if k != winlibnetwork.NetworkName && k != winlibnetwork.HNSID {
+						netOption[k] = v
+					}
+				}
+				name := v.Name()
+				id := v.ID()
+
+				err = v.Delete()
```
Comment (thaJeztah) [inline daemon/daemon_windows.go:279]: Just to be sure; would the `ingress` network also be in this list? If so, do we need to pass `NetworkDeleteOptionRemoveLB` as option? https://github.com/moby/moby/blob/5de77049db9d6aed893f086fc672cb8df4532650/libnetwork/network.go#L915

Comment (thaJeztah) [reply daemon/daemon_windows.go:279]: Oh, I guess the `ingress` network wouldn't be `v.Type() == "nat"` ? If that's so, ignore my comment

**Review by thaJeztah — APPROVED** [review]
LGTM

@evol262 perhaps you want to have a look as well?

File: daemon/daemon_windows.go (hunk 1 above)
Comment (olljanat) [reply daemon/daemon_windows.go:279]: `ingress` is `v.Type() == "overlay"` so this does not affect those.

Comment (thaJeztah) [reply daemon/daemon_windows.go:279]: Thanks! Yes, I realised that after writing my first comment; thanks for confirming 👍

**Review by evol262 — APPROVED** [review]

**Comment by thaJeztah** [comment]
alright, let's get this in!

**Pull request merged**